pkg/component/networking/coredns/coredns_test.go

```diff
@@ -1033,13 +1033,40 @@ status: {}
 				Expect(component.WaitCleanup(ctx)).To(Succeed())
 			})
 		})
+
+		var _ = Describe("ipFamilyPolicy", func() {
+			It("should return SingleStack if only IPv4 ClusterIP and both IP families requested", func() {
+				ipFamilies := []gardencorev1beta1.IPFamily{gardencorev1beta1.IPFamilyIPv4, gardencorev1beta1.IPFamilyIPv6}
+				clusterIPs := []net.IP{net.ParseIP("10.0.0.10")}
+				Expect(ipFamilyPolicy(ipFamilies, clusterIPs)).To(Equal(string(corev1.IPFamilyPolicySingleStack)))
+			})
+
+			It("should return PreferDualStack if both IPv4 and IPv6 ClusterIPs and both IP families requested", func() {
+				ipFamilies := []gardencorev1beta1.IPFamily{gardencorev1beta1.IPFamilyIPv4, gardencorev1beta1.IPFamilyIPv6}
+				clusterIPs := []net.IP{net.ParseIP("10.0.0.10"), net.ParseIP("2001:db8::1")}
+				Expect(ipFamilyPolicy(ipFamilies, clusterIPs)).To(Equal(string(corev1.IPFamilyPolicyPreferDualStack)))
+			})
+		})
```

Comment:
I am a bit surprised by this test because it tests the functions in this file and not the "real" ones. We should either test the real functions or remove this part of the test.